test: hash challenger test coverage

jorgeantonio21 opened the pull request:
Following #27, we add further test coverage for a hash challenger.

npwardberkeley reviewed (approved):
Nice work, thanks for contributing!

jorgeantonio21 commented:
Addressed the PR comments, for some reason the CI is failing while running `cargo fmt`. I checked on my side, and it seems that the files are correctly formatted with nightly version.

npwardberkeley commented:
Weird, I'm not sure quite where the difference comes from — seems like CI is running `fmt` with [group_imports](https://rust-lang.github.io/rustfmt/?version=v1.6.0&search=#group_imports) set to `StdExternalCrate` instead of `Preserve`. But I just changed it to satisfy the CI
